### PR TITLE
Declare a JSON schema in 'default' for each rule

### LINKED ISSUE #214: Use JSON schema `default` in rules
[Issuehunt badges](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/214)

Instead of manual `Object.assign`. See: https://github.com/eslint/eslint/commit/550de1e611a1e9af873bcb18d74cf2056e8d2e1b

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/214).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by Manu1400**
For fix #214 :
- [x] each rule have a JSON schema

### CODE REVIEW

File: rules/max-asserts.js [hunk 1]
```diff
@@ -56,12 +56,16 @@ const create = context => {
 };
 
 const schema = [{
-	type: 'integer'
+	title: 'maximum number of assertions for each test',
+	type: 'integer',
+	minimum: 0,
+	default: 5
```
Comment (sindresorhus) [inline rules/max-asserts.js]: When you define the defaults here, you also need to remove the default handling from the rule itself. For example: https://github.com/avajs/eslint-plugin-ava/blob/d812210ea6d21dba0aabfbbed1e7a60dbaec5c25/rules/max-asserts.js#L10

File: rules/max-asserts.js [hunk 2]
```diff
@@ -56,12 +56,16 @@ const create = context => {
 };
 
 const schema = [{
-	type: 'integer'
+	title: 'maximum number of assertions for each test',
```
Comment (sindresorhus) [inline rules/max-asserts.js]:
```suggestion
	title: 'Maximum number of assertions for each test',
```

File: rules/no-async-fn-without-await.js [hunk 3]
```diff
@@ -51,8 +51,10 @@ const create = context => {
 module.exports = {
 	create,
 	meta: {
+		type: 'problem',
 		docs: {
 			url: util.getDocsUrl(__filename)
-		}
+		},
+		schema: []
```
Comment (sindresorhus) [inline rules/no-async-fn-without-await.js]: No point in defining this property when it's just an empty array.

File: rules/no-import-test-files.js [hunk 4]
```diff
@@ -69,16 +69,28 @@ const create = context => {
 };
 
 const schema = [{
+	title: 'one option: files',
```
Comment (sindresorhus) [inline rules/no-import-test-files.js]: This is not a good title.

File: rules/no-import-test-files.js [hunk 5]
```diff
@@ -69,16 +69,28 @@ const create = context => {
...
+	$comment: 'the default value of option `files` is from package.json or ava.config.js',
```
Comment (sindresorhus) [inline rules/no-import-test-files.js]:
```suggestion
	$comment: 'The default value of the `files` option is from `package.json` or `ava.config.js`.',
```

File: rules/no-import-test-files.js [hunk 6]
```diff
@@ -69,16 +69,28 @@ const create = context => {
...
+			type: 'array',
+			minItems: 0,
+			uniqueItems: true,
+			items: {
+				type: 'string'
+			},
+			examples: [
+				['lib/**/*.test.js', 'utils/**/*.test.js']
+			],
+			default: null
```
Comment (sindresorhus) [inline rules/no-import-test-files.js]: `undefined` would be more correct.

File: rules/no-import-test-files.js [hunk 7]
```diff
@@ -69,16 +69,28 @@ const create = context => {
...
 		}
 	}
 }];
 
 module.exports = {
 	create,
+	type: 'problem',
```
Comment (sindresorhus) [inline rules/no-import-test-files.js]: ?

File: rules/no-identical-title.js [hunk 8]
```diff
@@ -62,8 +62,10 @@ const create = context => {
 module.exports = {
 	create,
 	meta: {
+		type: 'suggestion',
```
Comment (sindresorhus) [inline rules/no-identical-title.js]: This is actually a `problem` as AVA will not allow tests with identical titles.

**Comment by sindresorhus** [comment]
Bump :)

**Comment by Manu1400** [comment]
this PR is ready

File: rules/max-asserts.js [hunk 9]
```diff
@@ -5,9 +5,11 @@ const createAvaRule = require('../create-ava-rule');
 
 const notAssertionMethods = ['plan', 'end'];
 
+const MAX_NUMBER_ASSERTIONS_BY_DEFAULT = 5;
+
 const create = context => {
 	const ava = createAvaRule();
-	const maxAssertions = context.options[0] || 5;
+	const maxAssertions = context.options[0] || MAX_NUMBER_ASSERTIONS_BY_DEFAULT;
```
Comment (sindresorhus) [inline rules/max-asserts.js]: Why are you defining the default both here and in the schema?

File: rules/no-ignored-test-files.js [hunk 10]
```diff
@@ -94,6 +105,7 @@ module.exports = {
 		docs: {
 			url: util.getDocsUrl(__filename)
 		},
+		type: 'problem',
```
Comment (sindresorhus) [inline rules/no-ignored-test-files.js]: Should be placed above `docs`. Same with the other places.

File: rules/assertion-arguments.js [hunk 11]
```diff
@@ -141,14 +141,16 @@ const schema = [{
 			enum: [
 				'always',
 				'never'
-			]
+			],
+			default: null
```
Comment (sindresorhus) [inline rules/assertion-arguments.js]: `undefined` would be better

**Pull request closed without merging**